### PR TITLE
Data cleaning

### PR DESCRIPTION
**Pull request by karthikraja95**
Data Cleaning is done. Review the code and if everything seems good enough merge to master

### CODE REVIEW

File: data_cleaning.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	    if 6 in steps:
+	        modComm=re.sub(r'[^\s\w]+','',modComm)
+	    # Remove stop words
+	    if 7 in steps:
+	        words=modComm.split()
+	        new_words=[]
+	        for token in words:
+	            if token.strip() and token.strip().lower() not in stopwords:
+	                new_words.append(token)
+	        modComm= ' '.join(new_words)
```
Comment (Ashton-Sidhu) [inline data_cleaning.py]: If code is not being used it should be removed.

File: data_cleaning.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	        space_after = '({0})'.format(space_after) + r'\b'
+
+	        modComm = re.sub(space_before, r' \1', modComm)
+	        modComm = re.sub(space_after, r'\1 ', modComm)
+
+	        # dog's --> dog + 's
+	        modComm = re.sub(r'([a-zA-Z]+)(\'s)', r'\1 \2', modComm)
+
+	        # dogs' --> dogs '
+	        modComm = re.sub(r'([a-zA-Z]+)(\' )', r'\1 \2', modComm)
```
Comment (Ashton-Sidhu) [inline data_cleaning.py]: Same with this `if`, if this code is not being used - remove it.

File: data_cleaning.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+		description_lst=[]
+		word_lst=[]
+		for file_path in file_list:
+			with open(file_path, 'r', encoding='utf-8', errors='ignore') as file:
+				file_data = file.read()
+				file_data_new = self.data_preprocessing(file_data,steps=range(1, 7))
+				file_words = tokenizer.tokenize(str(file_data_new))
+				stemmer = SnowballStemmer("english")
+				wordnetLemmatizer=WordNetLemmatizer()
+				file_words=[stemmer.stem(wordnetLemmatizer.lemmatize(i)) for i in file_words if i not in stopwords]
```
Comment (Ashton-Sidhu) [inline data_cleaning.py]: For the tags themselves what is the plan for getting back the entire word?

File: data_cleaning.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+		tokenizer = RegexpTokenizer(r'\w+')
+		description_lst=[]
+		word_lst=[]
+		for file_path in file_list:
+			with open(file_path, 'r', encoding='utf-8', errors='ignore') as file:
+				file_data = file.read()
+				file_data_new = self.data_preprocessing(file_data,steps=range(1, 7))
+				file_words = tokenizer.tokenize(str(file_data_new))
+				stemmer = SnowballStemmer("english")
+				wordnetLemmatizer=WordNetLemmatizer()
```
Comment (Ashton-Sidhu) [inline data_cleaning.py]: Was removing adverbs, pronouns, verbs, etc considered? Should this option be tested for performance?

File: data_cleaning.py (hunk 1 above)
Comment (Ashton-Sidhu) [inline data_cleaning.py]: If code is not being used it should be removed.

File: data_cleaning.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+	    # Remove stop words
+	    if 7 in steps:
+	        words=modComm.split()
+	        new_words=[]
+	        for token in words:
+	            if token.strip() and token.strip().lower() not in stopwords:
+	                new_words.append(token)
+	        modComm= ' '.join(new_words)
+	    # clitics pattern, for space_before we need to add space before the matched pattern
+	    if 8 in steps:
```
Comment (Ashton-Sidhu) [inline data_cleaning.py]: Same with this if, if this code is not being used - remove it.

File: data_cleaning.py [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+	        modComm = modComm.lower()
+	    # Remove @ tags
+	    if 5 in steps:
+	        at_tags=' '.join(re.findall(r'(?<=@)[^\s]+\s?',modComm))
+	        modComm=re.sub(r'@.*?\s','',modComm)
+	    # Remove non alphanumeric characters except space
+	    if 6 in steps:
+	        modComm=re.sub(r'[^\s\w]+','',modComm)
+	    # Remove stop words
+	    if 7 in steps:
```
Comment (Ashton-Sidhu) [inline data_cleaning.py]: If code is not being used it should be removed.

**Comment by Ashton-Sidhu** [comment]
This pull request does not need an ipython notebook, please remove it.

File: data_cleaning.py [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+from nltk.sentiment.vader import SentimentIntensityAnalyzer
+from nltk.sentiment.util import * 
+from nltk import tokenize
+from nltk.tag import PerceptronTagger
+from nltk.data import find
+import string
+## Machine Learning
+import sklearn
+import sklearn.metrics as metrics
+
```
[the remark this answers is not in the document]
Comment (karthikraja95) [reply data_cleaning.py]: Sure will remove the the unused libraires

File: data_cleaning.py [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+DATA_DIR = "data"
+all_files = [pth for pth in Path(DATA_DIR).glob("description/*.txt") if pth.is_file()]
+
```
[the remark this answers is not in the document]
Comment (karthikraja95) [reply data_cleaning.py]: I will remove this and we can pass the directory of the files when we run the code.

File: data_cleaning.py [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+class DataCleaning():
+
+	
+	def data_preprocessing(self,text,steps=range(1, 7)):
+
+	    modComm = text
+	    # Remove html tags
+	    if 1 in steps:
+	        modComm = re.sub(r"\s*<.*?>",'', str(modComm))
```
[the remark this answers is not in the document]
Comment (karthikraja95) [reply data_cleaning.py]: I think we don't have any html tags, will remove those

File: data_cleaning.py [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+	    # Replace HTML character with ASCII 
+	    if 2 in steps:
+	        modComm = html.unescape(modComm)
+	    # Remove url in form of (http or https://)address.domain(/file)
+	    if 3 in steps:
+	        regex_url = r'\s*(?:https?:\/\/)?[\w.-]+(?:\.[\w.-]+)+[\w\-._~:/?#[\]@!\$&\'\(\)\*\+,;=.]+'
+	        modComm = re.sub(regex_url, '', modComm)
```
[the remark this answers is not in the document]
Comment (karthikraja95) [reply data_cleaning.py]: Yeah, thats a good suggestion. We will keep the urls then

File: data_cleaning.py (hunk 4 above)
Comment (karthikraja95) [reply data_cleaning.py]: We can implement part of speech to remove the verbs, adverbs etc, we will implement that feature

File: data_cleaning.py (hunk 3 above)
Comment (karthikraja95) [reply data_cleaning.py]: We have to think about it.

**Comment by karthikraja95** [comment]
I will submit another pull request with the modifications

**Pull request closed without merging**